### PR TITLE
NEXT-30083 - Documentation for Quotes

### PR DESCRIPTION
**Pull request by leduc92**

### CODE REVIEW

File: products/extensions/b2b-components/quotes-management/index.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+Partners have the freedom to accept or decline the offer, and upon acceptance, they are guided through the seamless checkout process. The system then automatically generates an order based on the accepted quote, optimizing efficiency and ensuring a smooth transition from negotiation to transaction. This feature revolutionizes B2B interactions by reducing the need for extensive manual discussions and fostering a more efficient and collaborative environment for both partners and merchants.
```
Comment (sushmangupta) [inline products/extensions/b2b-components/quotes-management/index.md:12]:
```suggestion
The Quote Management feature streamlines the B2B partnership process by enabling partners to seamlessly request and accept quotes without the need for time-consuming manual negotiations. The process begins with B2B partners populating their cart with desired products, after which they can initiate a quote request based on the contents of their cart. Once the request is submitted, B2B merchants have the ability to review the quote within the administration system. They can also apply discounts to individual product items within the quote to tailor the offer to the partner's needs. Subsequently, the modified quote is sent to the B2B partners for their consideration.

Partners are free to accept or decline the offer, and upon acceptance, they are guided through the seamless checkout process. The system then automatically generates an order based on the accepted quote, optimizing efficiency and ensuring a smooth transition from negotiation to transaction. This feature revolutionizes B2B interactions by reducing the need for extensive manual discussions and fostering a more efficient and collaborative environment for both partners and merchants.
```

File: products/extensions/b2b-components/quotes-management/quotes-conversion.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
+---
+nav:
+  title: Quotes conversion
+  position: 30
+
+---
+
+# Quotes conversion
+
+We allow customers to convert their shopping carts into quotes to facilitate seamless processing. We introduce 2 new services to handle the conversion process
```
Comment (sushmangupta) [inline products/extensions/b2b-components/quotes-management/quotes-conversion.md:10]:
```suggestion
Customers can convert their shopping carts into quotes to facilitate seamless processing. There are two new services to handle the conversion process :
```

File: products/extensions/b2b-components/quotes-management/quotes-conversion.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+**Cart to quote converter:**
```
Comment (sushmangupta) [inline products/extensions/b2b-components/quotes-management/quotes-conversion.md:12]:
```suggestion
## Cart to quote converter
```

File: products/extensions/b2b-components/quotes-management/quotes-conversion.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+When a customer wants to request a quote from their shopping cart, we basically convert a cart to an order, then process to enrich the data for quote. the method `convertToQuote` of class `Shopware\Commercial\B2B\QuoteManagement\Domain\CartToQuote\CartToQuoteConverter` is responsible for that process
```
Comment (sushmangupta) [inline products/extensions/b2b-components/quotes-management/quotes-conversion.md:14]:
```suggestion
When a customer wants to request a quote for their shopping cart, the process involves converting a cart to an order and then proceeding to enrich the data for the quote. The method `convertToQuote` of class `Shopware\Commercial\B2B\QuoteManagement\Domain\CartToQuote\CartToQuoteConverter` is responsible for this process.
```

File: products/extensions/b2b-components/quotes-management/quotes-conversion.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+        
+        //enrich quote line-items
+        
+        return $quote;
+    }
+```
+
+**Quote to cart converter:**
+
+When a customer wants to place an order based on a quote, we will create a new cart based on quote data, the method `convertToCart` of class `Shopware\Commercial\B2B\QuoteManagement\Domain\QuoteToCart\QuoteToCartConverter` is responsible for that process
```
Comment (sushmangupta) [inline products/extensions/b2b-components/quotes-management/quotes-conversion.md:37]:
```suggestion
When a customer wants to place an order based on a quote, a new cart is created based on the quote data. The method `convertToCart` of class `Shopware\Commercial\B2B\QuoteManagement\Domain\QuoteToCart\QuoteToCartConverter` is responsible for this process.
```

File: products/extensions/b2b-components/quotes-management/quotes-conversion.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+        
+        //enrich quote data
+        
+        //enrich quote line-items
+        
+        return $quote;
+    }
+```
+
+**Quote to cart converter:**
```
Comment (sushmangupta) [inline products/extensions/b2b-components/quotes-management/quotes-conversion.md:35]:
```suggestion
## Quote to cart converter
```

**Review by sushmangupta — CHANGES_REQUESTED** [review]
Please take a look at these suggestions.

**Review by sushmangupta — APPROVED** [review]

**Review by nguyenky — APPROVED** [review]

**Review by Isengo1989 — APPROVED** [review]

**Review by phuoccx — APPROVED** [review]

**Comment by RobinHoutevelts** [comment]
> When a customer wants to request a quote for their shopping cart, the process involves converting a cart to an order and then proceeding to enrich the data for the quote. The method `convertToQuote` of class `Shopware\Commercial\B2B\QuoteManagement\Domain\CartToQuote\CartToQuoteConverter` is responsible for this process.

@leduc92 Can someone point me to this class?
It doesn't seem to exist inside the `shopware/` organization when I [search for it on GitHub.](https://github.com/search?q=org%3Ashopware+CartToQuoteConverter&type=code)

**Comment by mitelg** [comment]
@RobinHoutevelts the described feature is part of the Commercial plugin, which is separated from the core. if you want to have access to that repo, you can ask our support for it

**Pull request merged**